sdks/python/apache_beam/pipeline.py

```diff
@@ -396,28 +400,57 @@ def replace_all(self, replacements):
     for override in replacements:
       self._check_replacement(override)
 
-  def run(self, test_runner_api=True):
-    """Runs the pipeline. Returns whatever our runner returns after running."""
-
+  def run(self, test_runner_api=True, runner=None, options=None,
```

Comment:
Why are we adding runner and options parameters here?

Reply:
IIRC, we want to allow the user to switch to `DataflowRunner` using the `p.run()` pattern instead of limiting the user to `Runner().run_pipeline(p, options)`.

Do you think we should put it into a separate PR, or simply not supporting it at all?

Reply:
> Do you think we should put it into a separate PR,

Yes. At least let's have a seperate discussion for API changes like this.

>  or simply not supporting it at all?

Maybe not. This could be just a override for the interactive runners run() (e.g. run_with(NewRunner, NewOptions). At least, let's discuss with all stakeholders.

Reply:
Putting this discussion on next Monday's agenda and will remove changes to the API.